pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go

```diff
@@ -73,8 +74,22 @@ func (r *PerconaServerMongoDBRestore) CheckFields() error {
 		return fmt.Errorf("spec clusterName field is empty")
 	}
 
-	if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
-		return errors.New("fields backupName or storageName and destination is empty")
+	if len(r.Spec.BackupName) == 0 && r.Spec.BackupSource == nil {
+		return errors.New("one of backupName or backupSource is required")
+	}
+
+	if r.Spec.BackupSource != nil {
+		if len(r.Spec.BackupSource.Destination) == 0 {
+			return errors.New("backupSource destination is required")
+		}
+
+		if r.Spec.BackupSource.Destination[:5] != "s3://" {
+			return errors.New("backupSource destination should use s3 protocol format")
+		}
+
+		if len(r.Spec.BackupSource.StorageName) == 0 && r.Spec.BackupSource.S3 == nil {
```

Comment:
should we also check if StorageNmae exists in the list of storages in Cluster CR?

Reply:
we check it in the `r.getStorage`: https://github.com/percona/percona-server-mongodb-operator/blob/2d7525b75ff10f2ad30bdb342d7e869f50c70cda/pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go#L327